### PR TITLE
Add support for Prime mice

### PR DESCRIPTION
**Pull request by sephiroth99**
This adds support for the SteelSeries Prime mice (#166).

Supported features:
- sensitivity
- led color
- led brightness
- polling rate

What is not supported:
- button mapping

By looking at the SSE3 database, it seems like the Prime and Prime+ support a new color handler. Instead, I was able to reuse the regular `color` handler by using the `command` and `command_suffix` fields to build a valid packet (but different from my captures). In short, the RGB color value is present twice in the data, at the start and at the end. Sending only at the start also works. But this means supporting the Prime+ will most likely require to correctly implement this new color handler.

There were a few changes to the core code, mainly:
- added support for variable size in the `range` handler (new option `range_length_byte`)
- added support for zero based indexing of the `multidpi_range` handler

### CODE REVIEW

**Comment by flozz** [comment]
> added support for zero based indexing of the multidpi_range handler

Hum that's annoying, I have already an implementation for this in the [aerox3-wireless](https://github.com/flozz/rivalcfg/blob/aerox3-wireless/rivalcfg/handlers/multidpi_range.py#L45) branch... Maybe I can merge this part of the branch on `master` to allow you to use it in your PR?

Except this ↑ your PR looks great :)

**Comment by sephiroth99** [comment]
> > added support for zero based indexing of the multidpi_range handler
> 
> Hum that's annoying, I have already an implementation for this in the [aerox3-wireless](https://github.com/flozz/rivalcfg/blob/aerox3-wireless/rivalcfg/handlers/multidpi_range.py#L45) branch... Maybe I can merge this part of the branch on `master` to allow you to use it in your PR?
> 
> Except this ↑ your PR looks great :)

I had a look at the `first_preset` option and it seems fine to me. If you want to put that part now in `master`, I'll rebase on top of it, no problem for me.

**Comment by flozz** [comment]
The updated `multidpi_range` is on `master` :)

**Comment by sephiroth99** [comment]
OK I updated the code to use the new `multidpi_range`, everything seem a-ok on my side! Let me know if there are any other issues.

File: rivalcfg/devices/prime.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                0x00,
+                0x00,
+                0x00,
+                0x00,
+                0xFF,
+            ],
+            "value_type": "rgbcolor",
+            "default": "#FF5200",
+        },
+        "wheel_brightness": {
```
Comment (flozz) [inline rivalcfg/devices/prime.py:72]: I think this should be renamed `led_brightness` to be consistent with other devices

**Review by flozz — CHANGES_REQUESTED** [review]
There is just a key to rename to be more consistent with other devices. Great job! :+1:

**Comment by flozz** [comment]
Thank you, the Prime support will be released in the next Rivalcfg version :)

**Pull request merged**